### PR TITLE
 Add .etlas file as possible configuration source

### PR DESCRIPTION
**Pull request by jneira**
* Tested almost all subcommands (including old ones)
* `etlas init` generates a `<pkgname>.etlas` file by default

### CODE REVIEW

**Comment by jneira** [comment]
i am afraid that `etlas old-install pkg.tar.gz` fails when it does contain a `.etlas` file instead a `.cabal` file

**Comment by rahulmutt** [comment]
@jneira Does `etlas old-install pkg.tar.gz` work now with your recent commits?

**Comment by jneira** [comment]
@rahulmutt just tested it again and it works fine
However i've find another ref to cabal file to manage, in https://github.com/typelead/etlas/blob/master/etlas/Distribution/Client/TargetSelector.hs#L1766.
I'll fix it asap.
(There is another one in `Distribution.Client.Patch` but i guess we are not going to patch packages with `.etlas`)

**Comment by jneira** [comment]
@rahulmutt all done!

File: etlas/Distribution/Client/Install.hs [hunk 1]
```diff
@@ -1369,15 +1369,17 @@ installLocalTarballPackage verbosity pkgid
...
+          dhallFile = absUnpackedPath </> "etlas.dhall"
       info verbosity $ (if isGit then "Copying " else "Extracting ")
                     ++ tarballPath
                     ++ " to " ++ tmpDirPath ++ "..."
       patchedExtractTarGzFile verbosity False tmpDirPath relUnpackedPath tarballPath patchesDir isGit isBinary
-      exists <- doesFileExist descFilePath
-      when (not exists) $
-        die' verbosity $ "Package .cabal file not found: " ++ show descFilePath
+      existsAny <- forM [descFilePath "etlas", descFilePath "cabal", dhallFile]
+                   (doesFileExist . descFilePath)
```
Comment (rahulmutt) [inline etlas/Distribution/Client/Install.hs]: Why is `descFilePath` called twice - once inside the first argument of `forM` and again in the second argument of `forM`?

**Review by rahulmutt — COMMENTED** [review]
Overall, incredible work @jneira! Had just one question about one of the changes - once that's resolved, I'll merge!

File: etlas/Distribution/Client/Install.hs (hunk 1 above)
Comment (jneira) [reply etlas/Distribution/Client/Install.hs]: Whoops, what a bug, good catch
I'll fix it asap

**Comment by jneira** [comment]
@rahulmutt i think it is fixed now

**Review by rahulmutt — APPROVED** [review]
Looks good now, thanks! Merging.

**Pull request merged**